src/exposition/http.rs

```diff
@@ -391,15 +330,15 @@ pub struct MetricMetadata {
 
 /// Contains a snapshot of metric readings.
 #[derive(Serialize, Deserialize)]
-pub struct Readings {
+pub struct MetricsSnapshot {
     datetime: DateTime<Utc>,
     unix_ns: u128,
-    counters: Vec<u64>,
-    gauges: Vec<i64>,
-    histograms: Vec<Option<Snapshot>>,
+    counters: Vec<(MetricMetadata, Option<u64>)>,
+    gauges: Vec<(MetricMetadata, Option<i64>)>,
```

Comment:
Can the value of these ever be `None`? I don't see a codepath where those would get inserted -- seems to me that if we wanted them, we would need to add an else branch to the `if let Some(value)` checks above?

Reply:
I thought for lazy gauges/counters they could be none. But I'm not seeing where that could be the case with the latest metriken version. @swlynch99 - curious about your thoughts on this.

Reply:
Yeah lazy counters and gauges can be `None` if the counter hasn't been used yet. (See https://github.com/iopsystems/metriken/blob/main/metriken/src/lazy.rs#L122).

You could also treat this the same as a metric being added dynamically and filter out all metrics whose values are `None`.

Reply:
Here value() always returns a u64 or i64 depending on the type, I don't see how to get an option anymore?

Reply:
Fixed.